kroxylicious-filter-test-support/src/main/java/io/kroxylicious/test/RequestFactory.java

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.test;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.EnumMap;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import org.apache.kafka.common.Uuid;
+import org.apache.kafka.common.acl.AclOperation;
+import org.apache.kafka.common.acl.AclPermissionType;
+import org.apache.kafka.common.message.ConsumerGroupDescribeRequestData;
+import org.apache.kafka.common.message.CreateAclsRequestData;
+import org.apache.kafka.common.message.CreateTopicsRequestData;
+import org.apache.kafka.common.message.DeleteAclsRequestData;
+import org.apache.kafka.common.message.DeleteGroupsRequestData;
+import org.apache.kafka.common.message.DeleteRecordsRequestData;
+import org.apache.kafka.common.message.DeleteTopicsRequestData;
+import org.apache.kafka.common.message.DescribeAclsRequestData;
+import org.apache.kafka.common.message.DescribeGroupsRequestData;
+import org.apache.kafka.common.message.InitProducerIdRequestData;
+import org.apache.kafka.common.message.LeaveGroupRequestData;
+import org.apache.kafka.common.message.ListOffsetsRequestData;
+import org.apache.kafka.common.message.MetadataRequestData;
+import org.apache.kafka.common.message.OffsetCommitRequestData;
+import org.apache.kafka.common.message.OffsetFetchRequestData;
+import org.apache.kafka.common.message.OffsetForLeaderEpochRequestData;
+import org.apache.kafka.common.message.ProduceRequestData;
+import org.apache.kafka.common.message.UpdateMetadataRequestData;
+import org.apache.kafka.common.protocol.ApiKeys;
+import org.apache.kafka.common.protocol.ApiMessage;
+import org.apache.kafka.common.resource.PatternType;
+import org.apache.kafka.common.resource.ResourceType;
+
+import io.kroxylicious.test.record.RecordTestUtils;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import info.schnatterer.mobynamesgenerator.MobyNamesGenerator;
+
+public class RequestFactory {
```

Comment:
(non blocking).. I'm not quite sure I see the value of this class.  If my test needs a request object, it cares about its shape.

Reply:
Hmm I see your point, but I feel like this is not the first time I've wanted to throw a bunch of different ApiKeys at a test to make sure they do something. I could imagine a test for the Multi Tenant filter that said I want messages for X,Y & Z ApiKeys and make sure we add the tenant ID as a header to each. Now that use cases might be even better served by message with the correct shape but its works with out, but I was wondering if this class could be supplied with custom builders which point it can service both use cases.

Moving in that later direction it would be better to move it to the Junit Extension (though that makes code generation hard, circular dependencies) and allow it to be a parameter source.